main.go

```diff
@@ -107,7 +107,6 @@ func main() {
 		setupLog.Error(err, "Unable to create controller", "controller", failedCtrl)
 		os.Exit(1)
```

Comment:
So one pattern I'm seeing in some operators is defining the webhook server explicitly: https://github.com/kubernetes-sigs/controller-runtime/blob/c83076e9f79258f23bcb200cf2391cf4ade939cf/pkg/manager/manager.go#L245-L248. I think it's intended for external / custom ones, but could be something to try for here. E.g.,

```go
	webhookServer := webhook.Server{
		CertDir:  "/tmp/k8s-webhook-server/serving-certs",
		CertName: "tls.crt",
		KeyName:  "tls.key",
	}

	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
		Scheme:                 scheme,
		MetricsBindAddress:     metricsAddr,
		Port:                   9443,
		HealthProbeBindAddress: probeAddr,
		LeaderElection:         enableLeaderElection,
		LeaderElectionID:       "14dde902.flux-framework.org",
		WebhookServer:          &webhookServer,
...
```
but that shouldn't cause the error we see here. Going to keep looking.

Reply:
I'm worried it's related to some deprecation https://kubernetes.io/docs/reference/kubernetes-api/extend-resources/validating-webhook-configuration-v1/.